packages/astro/src/core/build/static-build.ts

```diff
@@ -37,6 +37,8 @@ export interface StaticBuildOptions {
 
 const MAX_CONCURRENT_RENDERS = 10;
 
+const STATUS_CODE_PAGES = new Set(['/404', '/500']);
```

Comment:
Any chance you could pull in the changes from https://github.com/withastro/astro/pull/2525 to match all known status codes? I remember https://github.com/withastro/astro/issues/2195 was opened with that expectation.

```suggestion
const STATUS_CODE_PAGES = new Set(['/400', '/401', '/402', '/403', '/404', '/405', '/406', '/407', '/408', '/409', '/410', '/411', '/412', '/413', '/414', '/415', '/416', '/417', '/418', '/421', '/422', '/423', '/424', '/425', '/426', '/428', '/429', '/431', '/451', '/500', '/501', '/502', '/503', '/504', '/505', '/506', '/507', '/508', '/510', '/511']);
```

Reply:
I seem to recall some discussion about this and isn't the point of this feature because some hosts will route `/unknown-url` to `/404.html`? But they aren't, as far as I'm aware, routing to `/505`, etc. as those are pretty rare status code. Let me know if I'm wrong and Vercel or Netlify (or another) do this.

The concern would be that this prevents you from having a page that happens to be a 3 digit number.

Reply:
Not sure how to confirm their redirect behavior for these, but as https://github.com/withastro/astro/pull/2525 was merged a month ago I think it's fair to match that behavior.

Reply:
I don't think I agree with that, it was merged but that doesn't mean it's correct. What are the use-cases for these other status codes generating a `507.html`, etc? [Netlify](https://docs.netlify.com/routing/redirects/redirect-options/#custom-404-page-handling) and [Vercel](https://vercel.com/guides/custom-404-page) support 404.html but I don't see anything about other status codes. If there is another host that does this I'd b happy to make the change, but without that this seems like a feature for no one that causes other bugs.

Reply:
Fair enough! Go ahead and merge.